File: app/Option.cs
using Newtonsoft.Json.Linq;
class Option
{
    public static void AddCustomer()
    {
        try
        {
            Console.Write($"Id del Cliente: ");
            Customer customer = new Customer();
            customer.id = Console.ReadLine();
            Console.Write($"Nombre del Cliente: ");
            customer.firstName = Console.ReadLine();
            Console.Write($"Apellido del Cliente: ");
            customer.lastName = Console.ReadLine();
            Console.Write($"Teléfono del Cliente: ");
            customer.phone = Console.ReadLine();
            Console.Write($"Fecha de Nacimiento del Cliente(1990-01-01): ");
            try
            {
                customer.birthDate = DateTime.Parse(Console.ReadLine());
            }
            catch (System.Exception)
            {
                Console.Write($"{Environment.NewLine}Error: Formato de fecha incorrecto");
                return;
            }

            var success = Api.AddCustomer(customer);
            if (success) Console.Write($"{Environment.NewLine}Cliente agregado con éxito");
        }
        catch (Exception ex)
        {
            Console.Write(ex.Message);
            throw;
        }

    }

    public static void EditCustomer()
    {
        try
        {
            Console.Write($"Id del Cliente que desea editar: ");
            string id = Console.ReadLine();
            Customer customer = new Customer();
            customer.id = id;
            Console.Write($"Nombre del Cliente: ");
            customer.firstName = Console.ReadLine();
            Console.Write($"Apellido del Cliente: ");
            customer.lastName = Console.ReadLine();
            Console.Write($"Teléfono del Cliente: ");
            customer.phone = Console.ReadLine();
            Console.Write($"Fecha de Nacimiento del Cliente(1990-01-01): ");
            try
            {
                customer.birthDate = DateTime.Parse(Console.ReadLine());
            }
            catch (System.Exception)
            {
                Console.Write($"{Environment.NewLine}Formato de fecha incorrecto");
                return;
            }
            
            var success = Api.EditCustomer(id, customer);
            if (success) Console.Write($"{Environment.NewLine}Cliente editado con éxito");
        }
        catch (Exception ex)
        {
            Console.Write(ex.Message);
            throw;
        }

    }

    public static void DeleteCustomer()
    {
        Console.Write($"{Environment.NewLine}Digite el id del cliente: ");
        var id = Console.ReadLine();

        try
        {
            var success = Api.DeleteCustomer(id);

            if (success) Console.Write($"{Environment.NewLine}Cliente eliminado con éxito");
        }
        catch (System.Exception e)
        {
            Console.WriteLine(e.Message);
            throw;
        }
    }

    public static void GetCustomersOrderById()
    {
        List<Customer> customers = Api.GetCustomersList("id");

        try
        {
            for (var i = 0; i < customers.Count; i++)
            {
                Console.WriteLine("");
                JObject parsed = JObject.Parse(Newtonsoft.Json.JsonConvert.SerializeObject(customers[i]));

                foreach (var pair in parsed)
                {
                    Console.WriteLine("{0}: {1}", pair.Key, pair.Value);
                }
            }
        }
        catch (System.Exception e)
        {
            Console.WriteLine(e.Message);
            throw;
        }
    }

    public static void GetCustomersOrderByName()
    {
        List<Customer> customers = Api.GetCustomersList("name");

        try
        {
            for (var i = 0; i < customers.Count; i++)
            {
                Console.WriteLine("");
                JObject parsed = JObject.Parse(Newtonsoft.Json.JsonConvert.SerializeObject(customers[i]));

                foreach (var pair in parsed)
                {
                    Console.WriteLine("{0}: {1}", pair.Key, pair.Value);
                }
            }
        }
        catch (System.Exception e)
        {
            Console.WriteLine(e.Message);
            throw;
        }
    }

    public static void GetCustomersOrderByBirthDate()
    {
        List<Customer> customers = Api.GetCustomersList("birthDate");

        try
        {
            for (var i = 0; i < customers.Count; i++)
            {
                Console.WriteLine("");
                JObject parsed = JObject.Parse(Newtonsoft.Json.JsonConvert.SerializeObject(customers[i]));

                foreach (var pair in parsed)
                {
                    Console.WriteLine("{0}: {1}", pair.Key, pair.Value);
                }
            }
        }
        catch (System.Exception e)
        {
            Console.WriteLine(e.Message);
            throw;
        }
    }

    public static void GetCustomer()
    {
        Console.Write($"{Environment.NewLine}Digite el id del cliente: ");
        var option = Console.ReadLine();

        try
        {
            Customer customer = Api.GetCustomerById(option);

            Console.WriteLine("");
            JObject parsed = JObject.Parse(Newtonsoft.Json.JsonConvert.SerializeObject(customer));

            foreach (var pair in parsed)
            {
                Console.WriteLine("{0}: {1}", pair.Key, pair.Value);
            }
        }
        catch (System.Exception e)
        {
            Console.WriteLine(e.Message);
            throw;
        }
    }
}

File: app/Option.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.time.LocalDate;

public class Option {
    public static void addCustomer() throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        
        try {
            System.out.print("\nId del Cliente: ");
            Customer customer = new Customer();
            customer.setId(reader.readLine());
            System.out.print("Nombre del Cliente: ");
            customer.setFirstName(reader.readLine());
            System.out.print("Apellido del Cliente: ");
            customer.setLastName(reader.readLine());
            System.out.print("Teléfono del Cliente: ");
            customer.setPhone(reader.readLine());
            System.out.print("Fecha de Nacimiento del Cliente(1990-01-01): ");
            try {
                customer.setBirthDate(LocalDate.parse(reader.readLine()));
            } catch (Exception e) {
                System.out.print("\nError: Formato de fecha incorrecto");
                return;
            }

            boolean success = Api.addCustomer(customer);
            if (success) System.out.print("\nCliente agregado con éxito");

        } catch (Exception e) {
            System.out.println(e.getMessage());
            throw e;
        }
    }

    public static void editCustomer() throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        
        try {
            System.out.print("\nId del Cliente que desea editar: ");
            Customer customer = new Customer();
            String id = reader.readLine();
            customer.setId(id);
            System.out.print("Nombre del Cliente: ");
            customer.setFirstName(reader.readLine());
            System.out.print("Apellido del Cliente: ");
            customer.setLastName(reader.readLine());
            System.out.print("Teléfono del Cliente: ");
            customer.setPhone(reader.readLine());
            System.out.print("Fecha de Nacimiento del Cliente(1990-01-01): ");
            try {
                customer.setBirthDate(LocalDate.parse(reader.readLine()));
            } catch (Exception e) {
                System.out.print("\nError: Formato de fecha incorrecto");
                return;
            }

            boolean success = Api.editCustomer(id, customer);
            if (success) System.out.print("\nCliente editado con éxito");

        } catch (Exception e) {
            System.out.println(e.getMessage());
            throw e;
        }
    }

    public static void deleteCustomer() throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        
        try {
            System.out.print("\nDigite el id del cliente: ");
            String id = reader.readLine();

            boolean success = Api.deleteCustomer(id);
            if (success) System.out.print("{\nCliente eliminado con éxito");

        } catch (Exception e) {
            System.out.println(e.getMessage());
            throw e;
        }
    }

    public static void getCustomerById() throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        
        try {
            System.out.print("\nDigite el id del cliente: ");
            String id = reader.readLine();

            boolean success = Api.getCustomerById(id);
            //if (success) System.out.print("{\nCliente eliminado con éxito");

        } catch (Exception e) {
            System.out.println(e.getMessage());
            throw e;
        }
    }

    public static void getCustomersOrderById() {
        
        try {

            boolean success = Api.getCustomersList("id");
            //if (success) System.out.print("{\nCliente eliminado con éxito");

        } catch (Exception e) {
            System.out.println(e.getMessage());
            throw e;
        }
    }

    public static void getCustomersOrderByName() {
        
        try {

            boolean success = Api.getCustomersList("name");
            //if (success) System.out.print("{\nCliente eliminado con éxito");

        } catch (Exception e) {
            System.out.println(e.getMessage());
            throw e;
        }
    }

    public static void getCustomersOrderByBirthDate() {
        
        try {

            boolean success = Api.getCustomersList("birthDate");
            //if (success) System.out.print("{\nCliente eliminado con éxito");

        } catch (Exception e) {
            System.out.println(e.getMessage());
            throw e;
        }
    }
}
